ipatests/test_integration/test_cert.py

```diff
@@ -127,6 +128,84 @@ def test_ipa_getcert_san_aci(self):
         ipaddrs = ext.value.get_values_for_type(x509.IPAddress)
         assert ipaddrs == [ipaddress.ip_address(self.clients[0].ip)]
 
+    def test_getcert_list_profile(self):
+        """
+        Test that getcert list command displays the profile
+        for the requests
+        """
+        for requests in os.listdir(paths.CERTMONGER_REQUESTS_DIR):
+            result = self.master.run_command(
+                ["getcert", "list", "-i", requests], raiseonerr=False
+            )
+            assert "profile: caSubsystemCert" in result.stdout_text
+            assert "profile: caSignedLogCert" in result.stdout_text
+            assert "profile: caOCSPCert" in result.stdout_text
+            assert "profile: caSubsystemCert" in result.stdout_text
+            assert "profile: caCACert" in result.stdout_text
+            assert "profile: caServerCert" in result.stdout_text
+            assert "profile: caIPAserviceCert" in result.stdout_text
+            assert "profile: caIPAserviceCert" in result.stdout_text
+            assert "profile: KDCs_PKINIT_Certs" in result.stdout_text
```

Comment:
A couple of things. If this is really cycling through the requests then only one profile will be in each execution.

And it is not reliable that the filename will match the request id.

We're testing here whether certmonger displays the profile not that the profile is set for all IPA certs so I think you can just pick one and use that, something like:

getcert list -f HTTPD_CERT_FILE

It will be caIPAserviceCert

If you want to check them all then you'll need to collect something to query them on. You could collect them in a similar way that healthcheck does but that isn't very black box-like.

Otherwise, I think, you'll have to walk through CERTMONGER_REQUESTS_DIR collecting the id and then iterate on those and do a profile in [list of possibilities].

Reply:
Modified to check only specific cert e.g HTTPD_CERT_FILE